src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -8,9 +14,10 @@ import {
 } from '../../utils';
 import { Box } from '../Box';
 
-const Ref = React.forwardRef((props, ref) => (
-  <Box ref={ref}>{props.children}</Box>
-));
+const Ref = React.forwardRef((props, ref) => {
+  const { children } = props;
+  return <Box ref={ref}>{children}</Box>;
```

Comment:
This approach will not work. The Box will insert a `<div />` into the DOM, interrupting `<tr />` and `<li />` usage. The general approach was to have callers use the `ref` argument to the child render function as a way to avoid using `findDOMNode`.

Reply:
Just to clarify. I want to make sure this section of code receives ref?
```javascript
  items.slice(firstIndex, lastIndex + 1).forEach((item, index) => {
    const itemsIndex = firstIndex + index;
    // We only need page refs if we don't know the pageHeight
    // The new way, we pass the ref we want to the children render function.
    let ref;
    if (!pageHeight && itemsIndex === 0) {
      ref = firstPageItemRef;
    } else if (!pageHeight && (itemsIndex === step - 1 || itemsIndex === lastIndex)
    ) {
      ref = lastPageItemRef;
    } else if (show && show === itemsIndex) {
      ref = showRef;
    }

    let child = children(item, itemsIndex, ref);
```